polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -99,4 +111,23 @@ public boolean isChildOf(StorageLocation potentialParent) {
       return slashTerminatedLocation.startsWith(slashTerminatedParentLocation);
     }
   }
+
+  private static String scheme(String location) {
+    int schemePos = location.indexOf(':');
+    if (schemePos > 0) {
+      return location.substring(0, schemePos).toLowerCase(Locale.ROOT);
+    }
+    return null;
+  }
+
+  protected String normalizePath(String path) {
+    path = SLASHES.matcher(path).replaceAll("/");
```

Comment:
I don't think this is always correct. IIRC on Azure `/foo/bar` is not the same as `/foo//bar`

Reply:
It looks like this is just a special case for `file:///...` which should really just be treated as an empty authority

Reply:
Ahh yes, great catch. Can't believe I forgot about this but you're correct,  S3 is the same as well. I will make changes to make sure this is only happening for local files then.